lib/internal/assert/calltracker.js

```diff
@@ -59,7 +60,13 @@ class CallTracker {
         callChecks.add(context);
       }
       return ReflectApply(fn, this, arguments);
-    };
+    }
+
+    ObjectDefineProperty(callsFn, 'length', {
+      value: fn.length
```

Comment:
`%Object.prototype%.configurable`, `%Object.prototype%.enumerable`, and `%Object.prototype%.writable` could have been mutated in userland.
```suggestion
      configurable: true,​
      enumerable: false,
      value: 2​,
      writable: false,
```

Reply:
Ah yes, bad copy pasting sorry.

Alternatively, we could re-use the same property descriptor:

```js
ObjectDefineProperty(callsFn, 'length', ObjectGetOwnPropertyDescriptor(fn, 'length'));
```

Reply:
Get it, thanks, @targos  mentioned that Proxy is used for this interception, @aduh95  which solution do you prefer?

Reply:
Use whichever you prefer, I don't have a strong opinion on this as long as the tests pass. `Proxy` might be less error prone indeed.

Reply:
Ok, thanks very much